src/wallets/injected/SenderWallet.ts

```diff
@@ -114,6 +112,26 @@ class SenderWallet extends InjectedWallet implements ISenderWallet {
     return true;
   }
 
+  onAccountChanged() {
+    this.wallet.onAccountChanged((newAccountId) => {
+      console.log("SenderWallet:onAccountChange", newAccountId);
+
+      const state = getState();
+
+      this.wallet
+        .init({
+          contractId: state.options.accountId,
+        })
+        .then((response) => {
+          if (response.accessKey) {
+            this.emitter.emit("signIn");
+          } else {
+            this.signIn().then();
+          }
+        });
```

Comment:
I think we could reuse `this.signIn` here?

Reply:
You're right it can work by calling `this.signIn`  in at this point too. I was trying to figure out what to do  when a user closes the sender wallet window so I left that if else check but I see we're doing the same in `this.signIn`.